### DISCUSSION TITLE
Custom Script to Add Device to Existing or New Tenant

### DISCUSSION THREAD

**Discussion by paulcarre07** [netbox-community/netbox#17905] [Help Wanted!] [answered]
Hello everyone,

I would like to know if it’s possible to use a custom python script in NetBox to achieve the following, script without the API:

- Select an existing tenant, then create a device and add it to that tenant.
- If the specified tenant doesn’t already exist, provide an option to create a new tenant, then create the device and add it to the newly created tenant.

Any guidance on implementing this functionality would be greatly appreciated!

Thank you in advance for your help!

**Comment by jchambers2012** [comment, marked as answer]
The function you should use in your custom NetBox script is `get_or_create()`.  It returns a tuple of (object, created), where object is the retrieved or created object and created is a boolean specifying whether a new object was created.

Hope this helps!

## Example:
https://github.com/netbox-community/netbox/blob/4ca2b21a706f86fdb4ff5bf1940960f73ed74530/netbox/utilities/testing/utils.py#L47C1-L50C107
```
    manufacturer, _ = Manufacturer.objects.get_or_create(name='Manufacturer 1', slug='manufacturer-1')
    devicetype, _ = DeviceType.objects.get_or_create(model='Device Type 1', manufacturer=manufacturer)
    devicerole, _ = DeviceRole.objects.get_or_create(name='Device Role 1', slug='device-role-1')
    device = Device.objects.create(name=name, site=site, device_type=devicetype, role=devicerole, **attrs)
```

## Docs:
https://docs.djangoproject.com/en/5.1/ref/models/querysets/#get-or-create

**Reply by paulcarre07** [reply]
Thank you! I will check to see if this solution will work.
